src/commcare_cloud/commands/inventory_lookup/inventory_lookup.py

```diff
@@ -316,6 +316,7 @@ def run(self, args, unknown_args):
         remote_host_key, remote_port = self._SERVICES[args.service]
         loopback_address = f'127.0.{environment.terraform_config.vpc_begin_range}'
         remote_host = lookup_server_address(args.env_name, remote_host_key)
+        local_port = self.get_random_available_port()
```

Comment:
Noting that this is vulnerable to a race condition where some other process claims the port before the SSH connection+forward is established. Obviously it is impossible to report to the user (before establishing the connection) which port will be chosen if port selection is deferred to the SSH process by passing 0 in the port-forward directive. Not sure if it is possible to do it another way to avoid that issue?

Unfortunately I think the failure mode will not cause the entire SSH connection to fail (only the port forward will fail with `Could not request local forwarding.`), so unless you read the terminal output closely it may not be obvious.

Reply:
Yeah this is a good point. I thought of this, but didn't think it was likely in practice, though I could very much be overlooking how frequently OS processes ask for random ports and the chance of them overlapping. The ideal solution would have been if there was some way to tell SSH to bind to a random port and tell us what port it chose, but I couldn't find any mechanism in SSH -L that supported that. (When I tried, it appeared not to support `0` as a port option meaning choose a random port, in addition to the problem you mentioned of then not being able to report it to the user, but maybe I was doing something wrong.)